AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -301,6 +303,10 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
         stateMachine.notify(action: .notified)
     }
 
+    private func notifyFinished() {
+        mutationEventPublisher.send(completion: .finished)
```

Comment:
Why not include this in `finish()` (or include an invocation of this method from `finish()`? Are there cases when we shouldn't notify the publisher of completion when the operation finishes? If so, how are we cleaning up the sink & subscription in those cases?

Reply:
I tried overriding finish, but it is marked as final.. i guess we could call `finish() `within `notifyFinish()`, but this seems cleaner?